fix: allow dev command to start without entrypoint by creating defaul…

MAVRICK-1 opened the pull request:
This pull request introduces an improvement to the developer experience in the CLI registration flow. Now, if no entrypoint file is found, the CLI will automatically create a default `index.tsx` file with a simple board example, instead of just prompting the user to initialize the project manually.

Enhancements to entrypoint handling:

* If no entrypoint is detected, a default `index.tsx` file is created in the current working directory with a basic board component example, making it easier for users to get started.
* User messaging has been improved: the CLI now informs the user that a default entrypoint has been created and provides a tip about using `tsci init` for a full project setup



https://github.com/user-attachments/assets/d9c1dc46-edbf-4f49-8397-7d6143f68026



/claim #325
Fixes #325

MAVRICK-1 committed:
fix: allow dev command to start without entrypoint by creating default index.tsx

When no entrypoint is found, the dev command now creates a default index.tsx file
with a basic circuit example instead of failing. This improves the developer
experience for new projects.

Fixes #325

MAVRICK-1 committed:
fix: format code according to project style

seveibar reviewed (changes requested):
We shouldn't be creating an entrypoint, the code should just open runframe

MAVRICK-1 commented:
> We shouldn't be creating an entrypoint, the code should just open runframe

okay understood on it

MAVRICK-1 commented:
https://github.com/user-attachments/assets/a36f5101-492c-4ed8-9b20-f40c636cf7d2

@seveibar you are refering to this workflow ?

seveibar commented:
yes the user can select a file after the server starts. In my local project i have a ton of entrypoints that aren't being found and so the server doesn't start

MAVRICK-1 commented:
https://github.com/user-attachments/assets/7b4f0d7e-33a6-4ed5-9304-d3eb72f66c02

@seveibar like this right ?

seveibar commented:
yep

MAVRICK-1 committed:
fix: enable dev command to start without entrypoint in true sandbox mode

- Make DevServer componentFilePath optional (string | null)
- Allow CLI to start with no entrypoint files (filesystem or virtual)
- Start runframe in pure sandbox mode when no entrypoint exists
- Maintain backward compatibility with existing file-based workflows
- Fix all file operations to handle null componentFilePath gracefully

Fixes #325

MAVRICK-1 committed:
fix: correct TypeScript type for defaultMainComponentPath

Change null to undefined to match expected type signature

MAVRICK-1 commented:
@seveibar done , pushed the changes

imrishabh18 commented on cli/dev/register.ts:
```diff
@@ -53,20 +53,30 @@ export const registerDev = (program: Command) => {
           console.log(
-            "No entrypoint found. Run 'tsci init' to bootstrap a basic project.",
+            kleur.yellow(
+              "No entrypoint found. Starting development server in sandbox mode.",
```
```suggestion
              "No entrypoint found. Starting development server.",
```

Remove all the reference of `sandbox` from the other files too! It's not a sandbox

MAVRICK-1 committed:
Remove all sandbox references from dev command

- Updated user-facing message to remove "sandbox mode" terminology
- Updated code comments to use clearer language
- Changed "sandbox mode" to more descriptive alternatives

MAVRICK-1 committed:
Fix code formatting

Apply biome formatter fixes to maintain consistent code style.

MAVRICK-1 committed:
Merge remote-tracking branch 'origin/main' into fix/allow-dev-without-entrypoint

MAVRICK-1 commented:
@seveibar @imrishabh18 is it okay now?

seveibar commented on cli/dev/DevServer.ts:
```diff
@@ -271,17 +278,19 @@ export class DevServer {
-        postEvent("SNIPPET_SAVED")
-      },
-    })
+    if (this.componentFilePath) {
```
this should error instead of breaking silently

imrishabh18 reviewed (approved):
LGTM!

seveibar reviewed (changes requested):
CC @ShiboSoftwareDev after this change, automatic type installation will fail for some projects, just noting that an entrypoint is often not being found on my projects so i guess my auto type installation won't work, not a huge deal

MAVRICK-1 committed:
Fix: Make saveSnippet throw error when componentFilePath is null

Instead of silently failing when no component file path is specified,
the saveSnippet method now properly throws an error with a descriptive
message. This addresses the review feedback about not breaking silently.

MAVRICK-1 committed:
Merge remote-tracking branch 'origin/main' into fix/allow-dev-without-entrypoint

MAVRICK-1 commented:
@seveibar i dont know why the test are failing ? https://github.com/tscircuit/cli/pull/326/commits/1c8ba2224b038407befee2b8429794d0c70f1fa4 i have made this single change

MAVRICK-1 commented:
@seveibar just a heads-up
